pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -126,22 +126,49 @@ const (
 // If yes, you are expected to invalidate the cached predicate result for related API object change.
 // For example:
 // https://github.com/kubernetes/kubernetes/blob/36a218e/plugin/pkg/scheduler/factory/factory.go#L422
-
-// IMPORTANT NOTE: this list contains the ordering of the predicates, if you develop a new predicate
-// it is mandatory to add its name to this list.
-// Otherwise it won't be processed, see generic_scheduler#podFitsOnNode().
-// The order is based on the restrictiveness & complexity of predicates.
-// Design doc: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/predicates-ordering.md
 var (
+	// IMPORTANT NOTE: this list contains the ordering of the predicates, if you develop a new predicate
+	// it is mandatory to add its name to this list.
+	// Otherwise it won't be processed, see generic_scheduler#podFitsOnNode().
+	// The order is based on the restrictiveness & complexity of predicates.
+	// Design doc: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/predicates-ordering.md
 	predicatesOrdering = []string{CheckNodeConditionPred, CheckNodeUnschedulablePred,
 		GeneralPred, HostNamePred, PodFitsHostPortsPred,
 		MatchNodeSelectorPred, PodFitsResourcesPred, NoDiskConflictPred,
 		PodToleratesNodeTaintsPred, PodToleratesNodeNoExecuteTaintsPred, CheckNodeLabelPresencePred,
 		CheckServiceAffinityPred, MaxEBSVolumeCountPred, MaxGCEPDVolumeCountPred,
 		MaxAzureDiskVolumeCountPred, CheckVolumeBindingPred, NoVolumeZoneConflictPred,
 		CheckNodeMemoryPressurePred, CheckNodePIDPressurePred, CheckNodeDiskPressurePred, MatchInterPodAffinityPred}
+
+	// eCacheBypassPredicates bypasses some predicates which can not benefit a lot from cached predicates, including:
+	// 1. Cheap predicate: its calculation time is minor and fixed, i.e. its process time will not
+	// significantly increase with the growing of:
+	//   1. scale of the cluster;
+	//   2. complexity of candidate node, i.e. more labels, taints, annotations etc defined in node;
+	//   3. complexity of candidate pod, i.e. more labels, tolerations, affinities, volumes etc defined in pod;
+	//   4. complexity of the system, i.e. more services, workloads, namespaces etc in the Kubernetes cluster.
+	// 2. Other predicate: it can not be handled for some reason. But it is expected to be fixed in the future and removed
+	// from this list.
+	eCacheBypassPredicates = sets.NewString(
```

Comment:
Do we have data that eCache helps with other predicates. I know that I measured performance of eCache for inter-pod affinity and I know that it helps there, but I have a feeling that this list is not backed by numbers. Please let me know if that is not true.

Reply:
Or do you think it make more sense to hold this change for now, as the on-going eCache refactoring is coming?

Reply:
perfer to get the issue fixed; as someone said "Make It Work, Make It Right, Make It Fast" :)

Reply:
Sure, as we already have refactoring of eCache ready on dev branch. I think it's safe to hold this change, unless users complain.